azurerm/internal/services/datamigration/tests/resource_arm_data_migration_service_test.go

```diff
@@ -0,0 +1,228 @@
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataMigrationServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataMigrationService_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataMigrationServiceExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "subnet_id"),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku_name", "Standard_1vCores"),
+					resource.TestCheckResourceAttr(data.ResourceName, "kind", "Cloud"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDataMigrationService_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_migration_service", "test")
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataMigrationServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataMigrationService_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataMigrationServiceExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "subnet_id"),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku_name", "Standard_1vCores"),
+					resource.TestCheckResourceAttr(data.ResourceName, "kind", "Cloud"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.name", "test"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDataMigrationService_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_data_migration_service", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataMigrationServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataMigrationService_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataMigrationServiceExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMDataMigrationService_requiresImport),
+		},
+	})
+}
+
+func TestAccAzureRMDataMigrationService_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_migration_service", "test")
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataMigrationServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataMigrationService_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataMigrationServiceExists(data.ResourceName),
+				),
+			},
```

Comment:
can we add an import step here?

Reply:
Just wondering what is the rule we are following here 🤔. Is it like after every apply we should add a imoprt verfication?

Reply:
basically yes thou with a few exceptions. Every step should have an import to check the values in azure are as expected.

Reply:
As far as i know, for normal test step (instead of import test step), it will always verify that `Plan()` is empty right after each `Apply()` and we don't have a perpetual diff issue. It will do this with TWO plans. One without a refresh.
So is there anything more a import step will check?

Reply:
Yes, it will ensure that a read results in the same data being read into the schema - ie all values are being read back accurately